src/main/resources/com/google/api/codegen/ruby/initcode.snip

```diff
@@ -2,20 +2,7 @@
 
 @snippet initCode(initCodeSpec)
   @join line : initCodeSpec.lines
-    @switch line.lineType.toString
-    @case "StructureInitLine"
-      {@initLineStructure(line)}
-    @case "ListInitLine"
-      {@initLineList(line)}
-    @case "MapInitLine"
-      {@initLineMap(line)}
-    @case "SimpleInitLine"
-      {@initLineSimple(line)}
-    @case "ReadFileInitLine"
-      {@initLineReadFile(line)}
-    @default
-      $unhandledCase: {@line.lineType.toString}$
-    @end
+    {@initCodeLine(line)}
```

Comment:
Why does Ruby have this logic in two files (I just noticed!): `initcode.snip` and `standalone_sample.snip`? Is there anyway we can combine them into a SPOT?

(I'm fine doing this as a chaser PR, but if you're not doing it in this PR please file a issue so we remember)

Reply:
There are two helper methods to get a resource name in Ruby, which are handled by `initcode.snip`. Standalone samples prefer a different helper methods from in-code samples, and it's very hard to reuse the same `init_code` to generate them :(

Hopefully this will go away in the micro-generators, when we have a more idiomatic way to create resource names (from Earnest's design review)